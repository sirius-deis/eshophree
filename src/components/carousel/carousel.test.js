import { render, screen, fireEvent } from "@testing-library/react";
import Carousel from "./carousel";

const list = [
  <div key={1}>Slide 1</div>,
  <div key={2}>Slide 2</div>,
  <div key={3}>Slide 3</div>
];

describe("Carousel component", () => {
  it("should match snapshot", function () {
    const { container } = render(<Carousel>{list}</Carousel>);
    expect(container).toMatchSnapshot();
  });
  it("should render a carousel", () => {
    render(<Carousel>{list}</Carousel>);
    expect(screen.getByText("Slide 1")).toBeInTheDocument();
    expect(screen.getByText("Slide 2")).toBeInTheDocument();
    expect(screen.getByText("Slide 3")).toBeInTheDocument();
  })
  it("should navigate to the next slide", () => {
    render(<Carousel>{list}</Carousel>);
    const nextButton = screen.getByRole("button", { name: /›/i });
    fireEvent.click(nextButton);
    expect(screen.getByText("Slide 2").parentElement).toHaveClass("active");
    expect(screen.getByText("Slide 1").parentElement).not.toHaveClass("active");
  });
  it("should navigate to the previous slide on previous button click", () => {
    render(<Carousel>{list}</Carousel>);
    const nextButton = screen.getByRole('button', { name: /›/i });
    const prevButton = screen.getByRole('button', { name: /‹/i });
    fireEvent.click(nextButton);

    expect(screen.getByText("Slide 2").parentElement).toHaveClass("active");

    fireEvent.click(prevButton);
    expect(screen.getByText("Slide 1").parentElement).toHaveClass("active");
    expect(screen.getByText("Slide 2").parentElement).not.toHaveClass("active");
  });
});