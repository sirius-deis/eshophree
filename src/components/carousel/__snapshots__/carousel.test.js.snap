// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component should match snapshot 1`] = `
<div>
  <div
    class="sc-fFGjHI dwKuma"
  >
    <div
      class="sc-ewnqHT dLOUny"
    >
      <div
        style="display: flex; align-items: center; gap: 0.7rem;"
      >
        <div
          class="sc-gTRrQi kGWnip"
          color="text-color"
        />
      </div>
      <div>
        <button
          class="sc-gxYJeL sc-kPTPQs kCKvOC"
          disabled=""
        >
          ‹
        </button>
        <button
          class="sc-gxYJeL sc-ispOId kCKvOC"
          disabled=""
        >
          ›
        </button>
      </div>
    </div>
    <div
      class="sc-bgqQcB iPXChy"
      width="100"
    />
    <div
      style="display: flex; gap: 1rem;"
    >
      <div
        style="overflow: hidden;"
      >
        <div
          class="sc-dicizt cTmuUj"
          style="transform: translateX(0%);"
        >
          <div
            class="sc-iVCKna dnKQBn active"
          >
            <div>
              Slide 1
            </div>
          </div>
          <div
            class="sc-iVCKna dnKQBn"
          >
            <div>
              Slide 2
            </div>
          </div>
          <div
            class="sc-iVCKna dnKQBn"
          >
            <div>
              Slide 3
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
